lib/log_parser.rb

```diff
@@ -0,0 +1,32 @@
+class LogParser
+    def initialize(args)
+      if args.first.nil?
+        @file_name = "games.log"
+      else
+        @file_name = args.first
+      end
+    end
+
+    def execute
+      process_file(validate_file) unless validate_file.nil?
+    end
+  
+    def process_file (file)
+        file_data = file.readlines.map(&:chomp)
```

Comment:
Aqui vc nao precisa fazer um map em todas as linhas, pode retornar apenas a primeira

```suggestion
        file_data = file.readlines.first
```

Reply:
Concordo, já até tinha visto algumas sugestões, corrigido também.